core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregatorUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.batch.aggregator;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.aggregation.DeduplicateAggregationDefinition;
+import io.cdap.cdap.etl.api.relational.CoreExpressionCapabilities;
+import io.cdap.cdap.etl.api.relational.Expression;
+import io.cdap.cdap.etl.api.relational.ExpressionFactory;
+import io.cdap.cdap.etl.api.relational.Relation;
+import io.cdap.cdap.etl.api.relational.RelationalTranformContext;
+import io.cdap.cdap.etl.api.relational.StringExpressionFactoryType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Utility class for DedupAggregator.
+ */
+public class DedupAggregatorUtils {
+
+    public static DeduplicateAggregationDefinition generateAggregationDefinition
+            (RelationalTranformContext relationalTranformContext,
+             Relation relation,
+             DedupConfig.DedupFunctionInfo filterFunction,
+             List<String> uniqueFields) {
+        Optional<ExpressionFactory<String>> expressionFactory = relationalTranformContext.getEngine().
+                getExpressionFactory(StringExpressionFactoryType.SQL);
+        DeduplicateAggregationDefinition.FilterFunction aggFilterFunction = null;
```

Comment:
nit: you don't actually need `= null` as you do assignment in all branches

Reply:
yup right